codegen/smithy-ruby-rails-codegen-test/model/protocol-test/unions.smithy

```diff
@@ -248,6 +252,30 @@ apply JsonUnions @httpRequestTests([
             }
         }
     },
+    {
+        id: "RailsJsonSerializeRenamedStructureUnionValue",
```

Comment:
Is this something that we should contribute upstream?

Reply:
I ported these tests from the Smithy repo (I think they exist in most protocols).